### DISCUSSION TITLE
when / how to use config.beforeRedirect, i have no idea about this configuration field

### DISCUSSION THREAD

**Discussion by azhi09** [axios/axios#6152] [Q&A]
when / how to use `config.beforeRedirect`, i have no idea about this configuration field.
is this will work on `302` status?

**Comment by jasonsaayman** [comment]
`beforeRedirect` is a Node.js HTTP adapter option. It is called after Axios receives a redirect response, such as `301`, `302`, `303`, `307`, or `308`, and before Axios follows the next request.

It only runs when redirects are enabled, meaning `maxRedirects` is greater than `0`. It does not run in the browser adapters because browsers handle redirects themselves.

Example:

```js
await axios.get('https://example.com/start', {
  maxRedirects: 5,
  beforeRedirect: (options, { headers, statusCode }) => {
    console.log('redirect status:', statusCode);
    console.log('next hostname:', options.hostname);

    // You can inspect or modify the next request options here.
    // For example:
    // options.headers.Authorization = 'Bearer ...';
  }
});
```

So yes, it can run for a `302`, but only in Node.js and only when Axios is following that redirect.
